src/Avalonia.Controls/Utils/UndoRedoHelper.cs

```diff
@@ -91,6 +110,11 @@ public void Snapshot()
             }
         }
 
+        public void Clear()
+        {
+            if (!_undoRedoing) _states.Clear();
```

Comment:
Looks like a hack. Could we move this check to the TextBox itself?

Reply:
Done!